tests/kernel/gen_isr_table/src/main.c

```diff
@@ -385,7 +385,7 @@ static void *gen_isr_table_setup(void)
 #ifdef CONFIG_MULTI_LEVEL_INTERRUPTS
 ZTEST(gen_isr_table, test_multi_level_bit_masks_sec)
 {
-#if CONFIG_1ST_LEVEL_INTERRUPT_BITS < 10 && CONFIG_2ND_LEVEL_INTERRUPT_BITS < 10
+#if CONFIG_1ST_LEVEL_INTERRUPT_BITS < 10 || CONFIG_2ND_LEVEL_INTERRUPT_BITS < 10
```

Comment:
On this line itself  is the magic number 10. I can't comment on unchanged lines, but directly below this, there is the magic number 0x400. Further down, there are 0x40 and 0x800, 0x1, ..

Is there a way to make these tests pass for any valid selection of `CONFIG_1ST_LEVEL_INTERRUPT_BITS`, `CONFIG_2ND_LEVEL_INTERRUPT_BITS`, and / or `CONFIG_3RD_LEVEL_INTERRUPT_BITS`?

Reply:
I've updated the test to work regardless of the number of bits in each level